example/lib/main-search.dart

```diff
@@ -0,0 +1,367 @@
+    final left = offset.dx + index * emojiSpace + leftOffset;
+    final top = offset.dy;
+
+    // Generate other skintone options
+    final skinTonesEmoji = SkinTone.values
+        .map((skinTone) => EmojiPickerUtils().applySkinTone(emoji, skinTone))
+        .toList();
+
+    return OverlayEntry(
+      builder: (context) => Positioned(
+        left: left,
+        top: top,
+        child: Material(
+          elevation: 4.0,
+          child: EmojiContainer(
+            padding: const EdgeInsets.symmetric(vertical: 4.0),
+            color: _config.skinToneDialogBgColor,
+            buttonMode: _config.buttonMode,
+            child: Row(
+              children: [
+                _buildSkinToneEmoji(emoji, emojiSpace, emojiSize),
+                _buildSkinToneEmoji(skinTonesEmoji[0], emojiSpace, emojiSize),
+                _buildSkinToneEmoji(skinTonesEmoji[1], emojiSpace, emojiSize),
+                _buildSkinToneEmoji(skinTonesEmoji[2], emojiSpace, emojiSize),
+                _buildSkinToneEmoji(skinTonesEmoji[3], emojiSpace, emojiSize),
+                _buildSkinToneEmoji(skinTonesEmoji[4], emojiSpace, emojiSize),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  // Build Emoji inside skin tone dialog
+  Widget _buildSkinToneEmoji(
+    Emoji emoji,
+    double width,
+    double emojiSize,
+  ) {
+    return SizedBox(
+        width: width,
+        height: width,
+        child: EmojiCell.fromConfig(
+          emoji: emoji,
+          emojiSize: emojiSize,
+          onEmojiSelected: (category, emoji) {
+            _onEmojiSelected(category, emoji);
+            _closeSkinToneDialog();
+          },
+          config: _config,
+        ));
+  }
+
+  // Calucates the offset from the middle of selected emoji to the left side
+  // of the skin tone dialog
+  // Case 1: Selected Emoji is close to left border and offset needs to be
+  // reduced
+  // Case 2: Selected Emoji is close to right border and offset needs to be
+  // larger than half of the whole width
+  // Case 3: Enough space to left and right border and offset can be half
+  // of whole width
+  double _getLeftOffset(double emojiWidth, int column) {
+    var remainingColumns =
+        _config.columns - (column + 1 + (kSkinToneCount ~/ 2));
+    if (column >= 0 && column < 3) {
+      return -1 * column * emojiWidth;
+    } else if (remainingColumns < 0) {
+      return -1 *
+          ((kSkinToneCount ~/ 2 - 1) + -1 * remainingColumns) *
+          emojiWidth;
+    }
+    return -1 * ((kSkinToneCount ~/ 2) * emojiWidth) + emojiWidth / 2;
+  }
+
+  Widget _buildSearchBar(BuildContext context, bool isEmpty) {
+    return ColoredBox(
+      color: _config.bgColor,
+      child: Row(
+        crossAxisAlignment: CrossAxisAlignment.center,
+        children: [
+          if (!_isSearchFocused)
+            IconButton(
+              onPressed: _searchFocusNode.requestFocus,
```

Comment:
Not a blocker, but I would close the EmojiPicker when the search field is focused